src/ast.hpp

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <map>
```

Comment:
This one is a bit nitpick, but including map in here isn't really needed since we don't use map in header file.